### PR TITLE
NOX-574 Configure OpenApi Request and Response for Custom Route Mapping Targettng a Default Generated end point

### PR DESCRIPTION
**Pull request by anna-naumova25**

### CODE REVIEW

File: src/Nox.Solution/Models/Presentation/ApiConfiguration/ApiRouteMappings/ApiRouteMapping.cs [hunk 1]
```diff
@@ -47,12 +47,10 @@ public class ApiRouteMapping : YamlConfigNode<NoxSolution, ApiConfiguration>
     public ContentBodyType RequestBodyType { get; internal set; } = ContentBodyType.None;
 
     [IfEquals(nameof(RequestBodyType), ContentBodyType.Json)]
-    [Required]
     public JsonTypeDefinition JsonBodyType { get; internal set; } = null!;
```
Comment (rochar) [inline src/Nox.Solution/Models/Presentation/ApiConfiguration/ApiRouteMappings/ApiRouteMapping.cs]: make nullable explicit in such cases
```suggestion
    public JsonTypeDefinition? JsonBodyType { get; internal set; } = null!;
```

File: src/Nox.Solution/Models/Presentation/ApiConfiguration/ApiRouteMappings/ApiRouteMapping.cs [hunk 2]
```diff
@@ -47,12 +47,10 @@ public class ApiRouteMapping : YamlConfigNode<NoxSolution, ApiConfiguration>
...
 
-    [Required]
     [AdditionalProperties(false)]
-    public JsonTypeDefinition ResponseOutput { get; internal set; } = default!;
+    public JsonTypeDefinition ResponseOutput { get; internal set; } = null!;
```
Comment (rochar) [inline src/Nox.Solution/Models/Presentation/ApiConfiguration/ApiRouteMappings/ApiRouteMapping.cs]:
```suggestion
    public JsonTypeDefinition? ResponseOutput { get; internal set; } = null!;
```

File: src/Nox.Lib/Presentation/Api/Swagger/ApiRouteMappingDocumentFilter.cs [hunk 3]
```diff
@@ -78,39 +75,45 @@ public void Apply(OpenApiDocument openApiDocument, DocumentFilterContext context
...
             }
 
-            // add request body if it's supplied
-
-            if (route.JsonBodyType is not null 
-                && route.HttpVerb != HttpVerb.Get)
+            if (route.JsonBodyType is null && route.ResponseOutput is null)
             {
-                operation.RequestBody = new OpenApiRequestBody()
+                AddRequestBodyAndResponse(context, route, operation);
```
Comment (rochar) [inline src/Nox.Lib/Presentation/Api/Swagger/ApiRouteMappingDocumentFilter.cs]: Improve the method name to state what is doing, maybe :) :
```suggestion
                AddTargetEntityDefaultDocumentation(context, route, operation);
```

File: src/Nox.Lib/Presentation/Api/Swagger/ApiRouteMappingDocumentFilter.cs [hunk 4]
```diff
@@ -78,39 +75,45 @@ public void Apply(OpenApiDocument openApiDocument, DocumentFilterContext context
...
+            }
+            else
+            {
+                // add request body if it's supplied
```
Comment (rochar) [inline src/Nox.Lib/Presentation/Api/Swagger/ApiRouteMappingDocumentFilter.cs]: we could also extract this to a meaningful method making this function easier to read and understand?

File: src/Nox.Lib/Presentation/Api/RelatedEntityRouting/RelatedEntityRoutingPathBuilder.cs [hunk 5]
```diff
@@ -3,10 +3,17 @@
 using Nox.Solution;
 using Nox.Solution.Builders;
 using System.Text.RegularExpressions;
-using System.Linq;
 
 namespace Nox.Presentation.Api;
 
+internal sealed partial class OperationInfo
```
Comment (rochar) [inline src/Nox.Lib/Presentation/Api/RelatedEntityRouting/RelatedEntityRoutingPathBuilder.cs]: nice, this makes the code more easy to read and maintain, why partial?

Comment (rochar) [reply src/Nox.Lib/Presentation/Api/RelatedEntityRouting/RelatedEntityRoutingPathBuilder.cs]: this would fit well in a record?

**Review by rochar — CHANGES_REQUESTED** [review]
looks good, a couple of suggestions

**Comment by anna-naumova25** [comment]
@rochar  please see the code updated based on your comments.

**Review by rochar — APPROVED** [review]

**Pull request merged**